### PR TITLE
Pause the emulator based on mouse focus

### PR DESCRIPTION
**Pull request by gerikkub**
This was requested in issue #8232. Adds the option to pause the emulator when the mouse leaves the render window's focus. Defaults to false.

### CODE REVIEW

**Comment by skidau** [comment]
@dolphin-emu-bot rebuild

**Comment by skidau** [comment]
Could add a checkbox for this under Config > Interface

**Comment by MaJoRoesch** [comment]
"Added the ability to pause the emulator by moving the mouse outside the window"

So just moving the mouse away from the window makes the emulator pause? It doesn't even require clicking away? That doesn't sound like it would work very well for emulated wiimote users...

Why not have it pause when clicked away like most programs?

**Comment by gerikkub** [comment]
"Why not have it pause when clicked away like most programs?"

That seems like a better idea. I'll change it to pause when focus is moved away from the rendering window.

**Comment by skidau** [comment]
@dolphin-emu-bot rebuild

File: Source/Core/DolphinWX/ConfigMain.cpp [hunk 1]
```diff
@@ -486,6 +488,7 @@ void CConfigMain::InitializeGUITooltips()
 	ConfirmStop->SetToolTip(_("Show a confirmation box before stopping a game."));
 	UsePanicHandlers->SetToolTip(_("Show a message box when a potentially serious error has occurred.\nDisabling this may avoid annoying and non-fatal messages, but it may result in major crashes having no explanation at all."));
 	OnScreenDisplayMessages->SetToolTip(_("Display messages over the emulation screen area.\nThese messages include memory card writes, video backend and CPU information, and JIT cache clearing."));
+	PauseOnFocusLost->SetToolTip(_("Pauses the emulator when focus is taken away from the emulation screen."));
```
Comment (JosJuice) [inline Source/Core/DolphinWX/ConfigMain.cpp]: Emulation window, not emulation screen.

File: Source/Core/DolphinWX/Frame.cpp [hunk 2]
```diff
@@ -1176,6 +1176,35 @@ void CFrame::OnMouse(wxMouseEvent& event)
...
+		else
+		{
+			if (Core::GetState() == Core::CORE_RUN)
+			{
+				Core::SetState(Core::CORE_PAUSE);
+				if (SConfig::GetInstance().m_LocalCoreStartupParameter.bHideCursor)
+					m_RenderParent->SetCursor(wxNullCursor);
+				Core::UpdateTitle();
+			}
+			UpdateGUI();
```
Comment (lioncash) [inline Source/Core/DolphinWX/Frame.cpp]: `UpdateGUI()` can be moved out of this scope into the preceding one, since it's called in both cases.

**Comment by skidau** [comment]
@dolphin-emu-bot rebuild

File: Source/Core/DolphinWX/ConfigMain.cpp [hunk 3]
```diff
@@ -577,6 +580,7 @@ void CConfigMain::CreateGUIControls()
 	ConfirmStop = new wxCheckBox(DisplayPage, ID_INTERFACE_CONFIRMSTOP, _("Confirm on Stop"));
 	UsePanicHandlers = new wxCheckBox(DisplayPage, ID_INTERFACE_USEPANICHANDLERS, _("Use Panic Handlers"));
 	OnScreenDisplayMessages = new wxCheckBox(DisplayPage, ID_INTERFACE_ONSCREENDISPLAYMESSAGES, _("On-Screen Display Messages"));
+	PauseOnFocusLost = new wxCheckBox(DisplayPage, ID_INTERFACE_PAUSEONFOCUSLOST, _("Pause on Focus Lost"));
```
Comment (neobrain) [inline Source/Core/DolphinWX/ConfigMain.cpp]: This looks like broken English and is more technical than it needs to be. "Pause on (Window?) Focus Change" sounds more natural and intuitive.

Comment (bb010g) [inline Source/Core/DolphinWX/ConfigMain.cpp]: "Pause when Focus is Lost"?

Comment (neobrain) [inline Source/Core/DolphinWX/ConfigMain.cpp]: Sounds good, too, but "lost" should be spelt with lowercase `L`, I think.

Comment (JosJuice) [inline Source/Core/DolphinWX/ConfigMain.cpp]: The correct capitalization would be "Pause When Focus Is Lost". http://en.wikibooks.org/wiki/Basic_Book_Design/Capitalizing_Words_in_Titles

Comment (neobrain) [inline Source/Core/DolphinWX/ConfigMain.cpp]: @JosJuice There is no explicit language rule on things like this (the link just takes one arbitrary example set of rules). What matters is to moderately make sense and, more importantly, to be consistent, and I think having "lost" (being an adjective) lowercase here is consistent with the rest of our GUI.

Comment (JosJuice) [inline Source/Core/DolphinWX/ConfigMain.cpp]: Sure. I'm fine with that.

Comment (skidau) [inline Source/Core/DolphinWX/ConfigMain.cpp]: Maybe call the option "Pause On Inactive Window" ?

**Pull request merged**